api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -731,7 +731,7 @@ public User setUserActivationKey(User user) throws MessageException {
 		AdministrationService adminService = Context.getAdministrationService();
 		Locale locale = getDefaultLocaleForUser(user);
 		
-		String link = adminService.getGlobalProperty(OpenmrsConstants.GP_HOST_URL)
+		String link = adminService.getGlobalProperty(OpenmrsConstants.GP_PASSWORD_RESET_URL)
```

Comment:
For backwards compatibility, we could add some logic here to check if the new global property is empty but the old one has a value, we assign its value to the new global property. With a comment that this logic will need to be deleted once the deprecated property is eventually deleted.

Reply:
I added it.

Reply:
Is there any reason why you are reading the password reset url global property twice?

Reply:
Yes. After setting the value to `security.passwordResetUrl` the value of the reference `passwordResetGP` doesn't change. So It reads the value of security.passwordResetUrl twice.
If necessary, I could change the code by assigning the value to `passwordResetGP` from inside the if condition.